Sources/generate-pipeline/RuleCollector.swift

```diff
@@ -23,6 +23,10 @@ final class RuleCollector {
     /// The type name of the rule.
     let typeName: String
 
+    /// The description of the rule, extracted from the rule class or struct DocC comment
+    /// with `DocumentationCommentText(extractedFrom:)`
+    let description: String?
```

Comment:
Optional because `DocumentCommentText()` may return nil, but that's fine for us.